pykafka/balancedconsumer.py

```diff
@@ -535,7 +564,18 @@ def _rebalance(self):
 
                     # Only re-create internal consumer if something changed.
                     if new_partitions != self._partitions:
-                        self._setup_internal_consumer(list(new_partitions))
+                        cns = self._get_internal_consumer(list(new_partitions))
+                        old_offsets = _get_held_offsets(current_zk_parts, self._consumer)
```

Comment:
One last thing I just realised: this can now just be `old_offsets = self._consumer.held_offsets` (and something similar is true for `new_offsets`. That's also safer in case we recover after a zk session timeout, in which case maybe `current_zk_parts == {}`.

Other than that, I think this is all ready, and pretty awesome too.